ihmc-perception/src/main/java/us/ihmc/perception/gpuMapping/GpuMappingManager.java

```diff
@@ -233,8 +243,10 @@ public void destroy()
    {
       compressedHeightMapPointer.close();
       heightMapMessagePublisher.remove();
-      heightMapExtractor.destroy();
-      terrainMapExtractor.destroy();
+      if (processParameters.getPublishHeightMap())
+         heightMapExtractor.destroy();
+      if (processParameters.getPublishTerrainMap())
+         terrainMapExtractor.destroy();
```

Comment:
Wait a sec this doesn't seem good. The executors get created regardless of whether you're publishing the maps, so they should always get destroyed, right?


```suggestion
      heightMapExtractor.destroy();
      terrainMapExtractor.destroy();
```

Reply:
Mhmmmmmmmmmmmmmmm ya something is wrong here your right